src/backend/marsha/core/tests/test_api_video_share_live_media.py

```diff
@@ -0,0 +1,1190 @@
+                },
+            },
+        )
+        shared_live_media = SharedLiveMediaFactory(video=video)
+
+        jwt_token = AccessToken()
+        jwt_token.payload["resource_id"] = str(shared_live_media.video.id)
+        jwt_token.payload["roles"] = [random.choice(["instructor", "administrator"])]
+        jwt_token.payload["permissions"] = {"can_update": True}
+        jwt_token.payload["user"] = {"id": "56255f3807599c377bf0e5bf072359fd"}
+
+        with mock.patch.object(
+            api.video, "broadcast_message"
+        ) as mock_broadcast_message, mock.patch(
+            "marsha.core.serializers.xmpp_utils.generate_jwt"
+        ) as mock_jwt_encode:
+            mock_jwt_encode.return_value = "xmpp_jwt"
+
+            response = self.client.patch(
+                f"/api/videos/{shared_live_media.video.id}/start-sharing/",
+                HTTP_AUTHORIZATION=f"Bearer {jwt_token}",
+                data={"sharedlivemedia": str(shared_live_media.id)},
+                content_type="application/json",
+            )
+
+            self.assertEqual(response.status_code, 400)
+            self.assertEqual(response.json(), {"detail": "Video is already sharing."})
+            mock_broadcast_message.assert_not_called()
+
+    @override_settings(LIVE_CHAT_ENABLED=True)
+    @override_settings(XMPP_BOSH_URL="https://xmpp-server.com/http-bind")
+    @override_settings(XMPP_CONFERENCE_DOMAIN="conference.xmpp-server.com")
+    @override_settings(XMPP_DOMAIN="conference.xmpp-server.com")
+    @override_settings(XMPP_JWT_SHARED_SECRET="xmpp_shared_secret")
+    def test_api_video_shared_live_media_navigate_instructor(self):
+        """An instructor can navigate in a shared live media."""
+
+        shared_live_media = SharedLiveMediaFactory(nb_pages=6)
+        video = VideoFactory(
+            playlist__title="foo bar",
+            playlist__lti_id="course-v1:ufr+mathematics+00001",
+            active_shared_live_media=shared_live_media,
+            active_shared_live_media_page=1,
+            upload_state=PENDING,
+            live_state=RUNNING,
+            live_type=JITSI,
+            live_info={
+                "medialive": {
+                    "input": {
+                        "id": "medialive_input_1",
+                        "endpoints": [
+                            "https://live_endpoint1",
+                            "https://live_endpoint2",
+                        ],
+                    },
+                    "channel": {"id": "medialive_channel_1"},
+                },
+                "mediapackage": {
+                    "id": "mediapackage_channel_1",
+                    "endpoints": {
+                        "hls": {
+                            "id": "endpoint1",
+                            "url": "https://channel_endpoint1/live.m3u8",
+                        },
+                    },
+                },
+            },
+        )
+        video.shared_live_medias.set([shared_live_media])
```

Comment:
You also have to change `shared_live_media.video` I think ? Or what you made is enough ?

Reply:
`shared_live_media.video` is already set:
```
[30] > /app/src/backend/marsha/core/tests/test_api_video_share_live_media.py(603)test_api_video_shared_live_media_navigate_instructor()
-> jwt_token = AccessToken()
   7 frames hidden (try 'help hidden_frames')
(Pdb++) shared_live_media.video
<Video: Video 008>
```